File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if not len(s)  == len(t):
            return False
        
        map = [0] * 26
        
        for (a, b) in zip(s, t):
            if a == b:
                continue
                
            map[ord(a) - ord('a')] +=1;
            map[ord(b) - ord('a')] -=1;
            
        return all([n == 0 for n in map])

File: 242-valid-anagram/242-valid-anagram.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if (t.len() != s.len()){
            return false;
        }
        let mut s_arr = s.chars().collect::<Vec<char>>();
        s_arr.sort_unstable();
        
        let mut t_arr = t.chars().collect::<Vec<char>>();
        t_arr.sort_unstable();
        
        return s_arr == t_arr
    }
}